pkg/controller/node/node_controller.go

```diff
@@ -76,10 +77,14 @@ var _ reconcile.Reconciler = &ReconcileNode{}
 type ReconcileNode struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client client.Client
-	scheme *runtime.Scheme
+	client         client.Client
+	scheme         *runtime.Scheme
+	nmstateUpdater NmstateUpdater
 }
 
+// Added for test purposes
+type NmstateUpdater func(client client.Client, node *corev1.Node, namespace client.ObjectKey) error
+
```

Comment:
This will add some complexity only for unit test, but alternative are faking nmstatectl with at bash script and using FakeClient, but this also add a lot of complexity to unit test.

Reply:
after discussion it was decide to leave it this way for this PR